docs/pages/workflow/continuous-native-generation.mdx

```diff
@@ -25,91 +26,254 @@ Expo as a framework enables **Continuous Native Generation** by combining the fo
+
+There are cases where developers may want to swap between workflows often. For example, you may want to build custom functionality natively in Android Studio and Xcode, and then move that functionality into local config plugins.
 
-While library authors' adoption of CNG is not a prerequisite for using `npx expo prebuild`, it significantly enhances the user experience. If a library author has not yet adopted CNG, users can still use `npx expo prebuild` by creating local Config Plugins to modify the native generation pipeline. This flexibility ensures that the adoption of CNG is accessible and beneficial to all users within the React Native community.
+## Templates
 
-## Success
+You can customize how the native folders are generated by [config plugins](/config-plugins/introduction). Many config plugins already exist for lots of modifications, and community libraries often ship their own as well. You can [see a list of some popular plugins](https://github.com/expo/config-plugins) for more information.
 
-CNG is designed to empower developers to build better apps than they normally could, by dividing configuration into testable and maintainable pieces. This allows developers to build more powerful apps without compromising on iteration speed.
+Prebuild starts from template files which are then modified with config plugins. The template files are based on the Expo SDK version and come from the npm package [`expo-template-bare-minimum`](https://github.com/expo/expo/tree/main/templates/expo-template-bare-minimum). You can change the template used by passing `--template /path/to/template.tgz` to the `npx expo prebuild` command. This is not generally recommended because the base modifiers in `@expo/prebuild-config` make some undocumented assumptions about the template files, so it may be tricky to maintain your custom template.
 
-This pattern has been proven to work in the React Native community in the following ways.
+> **Note:** In network environments where all packages are downloaded from a private registry and npm public registry access is blocked, a locally-available template must be passed to the prebuild command. [Learn more about using a local version of the default template](https://expo.fyi/prebuild-without-npm-access).
 
-### Upgrades
+## Side effects
+
+`npx expo prebuild` performs several side effects outside of generating the **android** and **ios** directories. Work is in progress to eliminate these side effects &mdash; ideally, running `npx expo prebuild` would generate the Android and iOS projects and leave the rest of the project untouched.
+
+In addition to generating the native folders, prebuild also makes the following modifications:
+
+- Modifies the `scripts` field in the **package.json**
```

Comment:
would be good to describe what is change is. I thought it was changing the ios/android commands to `expo run:ios|android` or something?

Reply:
You are right. It does adds `expo run` commands and replaces the `start` command in the default template.

![CleanShot 2024-12-16 at 17 54 24@2x](https://github.com/user-attachments/assets/d236ad93-4ad7-4b90-815f-011059c9c58b)

Reply:
description added.